hooks/pre_validations/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,54 @@
+MONGO_DIR = '/var/lib/mongodb/'
+scenario = YAML.load_file('./config/last_scenario.yaml').fetch(:answer_file)
+
+# Pull MongoDB connection info from answer file
+if scenario = 'katello'
+  MONGO_HOST = param('katello', 'pulp_db_seeds').value
+else
+  MONGO_HOST = param('foreman_proxy_content', 'pulp_db_seeds').value
+end
+
+def fail_and_exit(message)
+  Kafo::Helpers.log_and_say :error, message
+  kafo.class.exit 1
+end
+
+def disk_space
+  # Check diskspace in /var/tmp
+  logger.info 'Checking available diskspace in /var/tmp for upgrade'
+  total_space = `df -k --output=avail /var/tmp`.split("\n").last.to_i
+  mongo_size = File.directory?(MONGO_DIR) ? `du -s  #{mongo_dir}`.split[0].to_i : 0
+  if total_space < mongo_size
+    logger.error "There is not enough free space #{total_space}, the size of MongoDB database is #{mongo_size}, please add additional space to /var/tmp and try again, exiting."
+    kafo.class.exit(1)
+  else
+    logger.debug "There is #{total_space} free space on disk, which is more than the size of the MongoDB database of #{mongo_size}, continuing with upgrade"
+  end
+end
+
+if app_value(:upgrade_mongo_storage_engine)
+  # Fail if MongoDB is already running in wiredTiger.
+  if File.file?("#{MONGO_DIR}/WiredTiger.wt")
+    fail_and_exit 'MongoDB has already been switched to wiredTiger. Exiting.'
+  end
+
+  # Fail if MongoDB is not localhost
+  if MONGO_HOST != 'localhost:27017'
+    fail_and_exit 'Upgrading is not supported on remote MongoDB database connections'
+  end
+
+  # Fail if we are not using the Katello/Foreman Proxy Content scenario.
+  katello = Kafo::Helpers.module_enabled?(@kafo, 'katello')
+  foreman_proxy_content = Kafo::Helpers.module_enabled?(@kafo, 'foreman_proxy_content')
+  fail_and_exit 'MongoDB storage engine upgrade is not currently supported for the chosen scenario.' unless katello || foreman_proxy_content
+
+  # Verify to stop services before starting migration
+  response = ask('Are you sure you want to continue? This will stop all the pulp services and will take a while depending on the size of your MongoDB database.')
```

Comment:
Totally agree, requiring user input means I have to use expect to do an automated/scripted upgrade. I do not want to use expect.

To add we don't do this for any previous migration (Puppet, qpidd, ect..) so I don't think we should start now.

Reply:
Fair point, I will remove it.